psi-java/src/main/kotlin/io/spine/tools/psi/java/TrimmingPsiFactory.kt

```diff
@@ -0,0 +1,164 @@
+ */
+
+package io.spine.tools.psi.java
+
+import com.intellij.pom.java.LanguageLevel
+import com.intellij.psi.PsiAnnotation
+import com.intellij.psi.PsiCodeBlock
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementFactory
+import com.intellij.psi.PsiEnumConstant
+import com.intellij.psi.PsiExpression
+import com.intellij.psi.PsiField
+import com.intellij.psi.PsiJavaCodeReferenceElement
+import com.intellij.psi.PsiJavaModule
+import com.intellij.psi.PsiMethod
+import com.intellij.psi.PsiParameter
+import com.intellij.psi.PsiRecordHeader
+import com.intellij.psi.PsiResourceVariable
+import com.intellij.psi.PsiStatement
+import com.intellij.psi.PsiType
+import com.intellij.psi.PsiTypeElement
+import com.intellij.psi.PsiTypeParameter
+
+/**
+ * Wraps the provided [PsiElementFactory] to allow the passed text representation
+ * of Java elements contain leading and trailing whitespaces.
+ *
+ * This decorator performs a preliminary trimming of the passed Java text for all
+ * `create...FromText()` methods that do not do it on their own. This is needed
+ * because we often use multiline Kotlin strings to define text, and when being
+ * interpolated, the default [trimIndent] usually does nothing to the string literal,
+ * leaving all whitespaces as is. The original PSI factory often throws in such cases.
+ *
+ * For example, the code snippet below:
+ * ```
+ * val statements = """
+ *     System.out.println("Hello world!");
+ *     System.out.println("My name is Jack.");
+ * """.trimIndent()
+ *
+ * val main = """
+ *     public static void print() {
+ *         $statements
+ *     }
+ * """.trimIndent()
+ * ```
+ *
+ * Leads to the following content of the `main` variable:
+ * ```
+ *     public static void print() {
+ *         System.out.println("Hello world!");
+ * System.out.println("My name is Jack.");
+ *     }
+ * ```
+ *
+ * Instead of the expected (notice the lack of leading whitespaces):
+ * ```
+ * public static void print() {
+ *     System.out.println("Hello world!");
+ *     System.out.println("My name is Jack.");
+ * }
+ * ```
+ *
+ * We are not afraid to lose in terms of formatting because of the following:
+ *
+ * 1. Formatting usually goes somewhere inside the element definition rather before.
+ * 2. At the last stage, the generated code is usually auto-formatted.
+ *
+ * The methods below are not overridden because they trim on their own:
+ *
+ * 1. [PsiElementFactory.createDocTagFromText].
+ * 2. [PsiElementFactory.createDocCommentFromText].
+ * 4. [PsiElementFactory.createModuleStatementFromText].
+ * 5. [PsiElementFactory.createModuleReferenceFromText].
+ *
+ * [PsiElementFactory.createClassFromText] also left intact because the created class is nested
+ * into outer `Dummy` class. So, the passed whitespaces go to definition of `Dummy`.
+ */
+@Suppress("TooManyFunctions")
+internal class TrimmingPsiFactory(
```

Comment:
The name suggests we're building a barbershop context... :) 

Please think of a name that would allow to extend this interface with other facilities not related to trimming.